File: ATL.Prototype/BLL/Settings/Menu.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Model.Settings;

namespace BLL.Settings
{
    public class Menu
    {
        DAL.Settings.Menu _menu = new DAL.Settings.Menu();

        public string GetData()
        {
            List<MenuEntity> list = _menu.GetData();
            StringBuilder psb = new StringBuilder();
            string cate;
            foreach (var m in list)
            {
                psb.Append("{");
                psb.Append("id");
                psb.Append(":");
                psb.Append(m.ID);
                psb.Append(",pId:");
                psb.Append(m.PID);
                psb.Append(",name:\"");
                psb.Append(String.IsNullOrEmpty(m.NAME) ? "" : m.NAME);
                psb.Append("\",title:\"");
                psb.Append(String.IsNullOrEmpty(m.TITLE) ? "" : m.TITLE);
                psb.Append("\",url");
                psb.Append(":\"");
                psb.Append(String.IsNullOrEmpty(m.URL) ? "" : m.URL);
                psb.Append("\",open");
                psb.Append(":\"");
                psb.Append(String.IsNullOrEmpty(m.OPEN) ? "" : m.OPEN);
                psb.Append("\",target");
                psb.Append(":\"");
                psb.Append(String.IsNullOrEmpty(m.TARGET) ? "_self" : m.TARGET);
                psb.Append("\",icon");
                psb.Append(":\"");
                psb.Append(String.IsNullOrEmpty(m.ICON) ? "" : m.ICON);

                psb.Append("\",sort");
                psb.Append(":\"");
                psb.Append(m.SORT);

                psb.Append("\",pname");
                psb.Append(":\"");
                psb.Append(m.PNAME);

                psb.Append("\"},");
            }


            cate = "[";

            cate += psb.ToString().TrimEnd(',');
            cate += "]";
            return cate;
        }
        public string GetDataNoRoot()
        {
            List<MenuEntity> list = _menu.GetDataNoRoot();
            StringBuilder psb = new StringBuilder();
            string cate;
            foreach (var m in list)
            {
                psb.Append("{");
                psb.Append("id");
                psb.Append(":");
                psb.Append(m.ID);
                psb.Append(",pId:");
                psb.Append(m.PID);
                psb.Append(",name:\"");
                psb.Append(String.IsNullOrEmpty(m.NAME) ? "" : m.NAME);
                psb.Append("\",title:\"");
                psb.Append(String.IsNullOrEmpty(m.TITLE) ? "" : m.TITLE);
                psb.Append("\",url");
                psb.Append(":\"");
                psb.Append(String.IsNullOrEmpty(m.URL) ? "" : m.URL + "?mid=" + m.ID);
                psb.Append("\",open");
                psb.Append(":\"");
                psb.Append(String.IsNullOrEmpty(m.OPEN) ? "" : m.OPEN);
                psb.Append("\",target");
                psb.Append(":\"");
                psb.Append(String.IsNullOrEmpty(m.TARGET) ? "_self" : m.TARGET);
                psb.Append("\",icon");
                psb.Append(":\"");
                psb.Append(String.IsNullOrEmpty(m.ICON) ? "" : m.ICON);

                psb.Append("\",sort");
                psb.Append(":\"");
                psb.Append(m.SORT);

                psb.Append("\",pname");
                psb.Append(":\"");
                psb.Append(m.PNAME);

                psb.Append("\"},");
            }


            cate = "[";

            cate += psb.ToString().TrimEnd(',');
            cate += "]";
            return cate;

        }
        public int Add(MenuEntity entity)
        {
            return _menu.Add(entity);
        }
        public int Edit(MenuEntity entity)
        {
            return _menu.Edit(entity);
        }
        public int Delete(decimal ID)
        {
            return _menu.Delete(ID);
        }
    }
}

File: ATL.Prototype/Web/Settings/Menu.js
var addOrEdit = 'edit';
$(function () {
    Init_dialog_PID();
    InitMenu();
    $('#PID').click(function () {
        $('#dialog_PID').dialog('open');
    });
    $('#add').click(function () {
        $('#add').click(function () {
            $('#Table_menu input[type=text]').val('');
            $('#Table_menu textarea').val('');
            $('#TARGET').val('_self');
            $('#OPEN').val('true');
            addOrEdit = 'add';
        });
    });
    $('#save').click(function () {
        if (addOrEdit == 'add') {
            Add();
        } else if (addOrEdit == 'edit') {
            Edit();
        }
    });
    $('#del').click(function () {
        $.messager.confirm('消息', '确认删除？', function (r) {
            if (r) {
                Del();
            }
        });
    });
});

function InitMenu() {
    $.getJSON('/api/Menu/GetData',
                   function (data) {
                       $.fn.zTree.init($("#menu"), {
                           view: {
                               showIcon: false,
                               showLine: true,
                               showTitle: true
                           },
                           data: {
                               simpleData: {
                                   enable: true
                               },
                               key: {
                                   title: "title"
                               }
                           },
                           callback: {
                               beforeClick: beforeClick,
                               onClick: onClick
                           }
                       }, eval(data));
                   });
    $.getJSON('/api/Menu/GetData',
        function (data) {
            $.fn.zTree.init($("#tree_Doc_MNG_PID"), {
                view: {
                    showIcon: false,
                    showLine: true,
                    showTitle: true
                },
                data: {
                    simpleData: {
                        enable: true
                    },
                    key: {
                        title: "title"
                    }
                },
                callback: {
                    beforeClick: beforeClick_pid,
                    onClick: onClick_pid
                }
            }, eval(data));
        });
}

function beforeClick(treeId, treeNode, clickFlag) {
    $('#menu a').removeAttr('href');
    addOrEdit = 'edit';
    return (treeNode.click != false);
}
function onClick(event, treeId, treeNode, clickFlag) {

    $('#ID').val(treeNode.id);
    $('#PID').val(treeNode.pname);
    $('#PID').attr('pId', treeNode.pId);
    $('#NAME').val(treeNode.name);
    $('#TITLE').val(treeNode.title);
    $('#URL').val(treeNode.url);
    $('#SORT').val(treeNode.sort);
    $('#TARGET').val(treeNode.target == '_self' ? '_self' : '_blank');
    $('#OPEN').val(treeNode.open == true ? 'true' : 'false');    
}

function beforeClick_pid(treeId, treeNode, clickFlag) {
    $('#tree_Doc_MNG_PID a').removeAttr('href');
    return (treeNode.click != false);
}
function onClick_pid(event, treeId, treeNode, clickFlag) {
    $('#PID').val(treeNode.name);
    $('#PID').attr('pid', treeNode.id);
    $('#dialog_PID').dialog('close');
}
function Init_dialog_PID() {
    $('#dialog_PID').dialog({
        width: 300,
        height: 400,
        closed: true
    });
}

function Add() {
    if ($('#PID').attr('pid') == null ||
        $('#NAME').val() == null) {
        $.messager.show({
            title: '消息',
            msg: '请填写必填信息',
            showType: 'show'
        });
        return;
    }
    var j = {
        'PID': $('#PID').attr('pId'),
        'NAME': $('#NAME').val(),
        'TITLE': $('#TITLE').val(),
        'ICON': null,
        'URL': $('#URL').val(),
        'OPEN': $('#OPEN').val(),
        'TARGET': $('#TARGET').val(),
        'SORT': $('#SORT').val()
    };
    $.ajax({
        type: 'post',
        url: '/api/Menu/Add',
        data: JSON.stringify(j),
        dataType: 'json',
        contentType: 'application/json',
        success: function (data) {
            if (data > 0) {
                $.messager.show({
                    title: '消息',
                    msg: '成功',
                    showType: 'show'
                });
                InitMenu();
            } else if (data == 0) {
                $.messager.show({
                    title: '消息',
                    msg: '重复',
                    showType: 'show'
                });
            }
        }
    });
}

function Edit() {
    if ($('#PID').attr('pId') == null ||
        $('#NAME').val() == null) {
        $.messager.show({
            title: '消息',
            msg: '请填写必填信息',
            showType: 'show'
        });
        return;
    }
    var j = {
        'ID': $('#ID').val(),
        'PID': $('#PID').attr('pId'),
        'NAME': $('#NAME').val(),
        'TITLE': $('#CATEGORY_DESC').val(),
        'ICON': null,
        'URL': $('#URL').val(),
        'OPEN': $('#OPEN').val(),
        'TARGET': $('#TARGET').val(),
        'SORT': $('#SORT').val()
    };
    
    $.ajax({
        type: 'post',
        url: '/api/Menu/Edit',
        data: JSON.stringify(j),
        dataType: 'json',
        contentType: 'application/json',
        success: function (data) {
            if (data > 0) {
                $.messager.show({
                    title: '消息',
                    msg: '成功',
                    showType: 'show'
                });
                InitMenu();
            } else if (data == 0) {
                $.messager.show({
                    title: '消息',
                    msg: '重复',
                    showType: 'show'
                });
            }
        }
    });
}

function Del() {
    if ($('#ID').val() == null) {
        $.messager.show({
            title: '消息',
            msg: '请选择目录',
            showType: 'show'
        });
        return;
    }
    $.ajax({
        type: 'get',
        url: '/api/Menu/Delete?ID=' + $('#ID').val(),
        dataType: 'json',
        success: function (data) {
            if (data > 0) {
                $.messager.show({
                    title: '消息',
                    msg: '成功',
                    showType: 'show'
                });
                InitMenu();
            } else {
                $.messager.show({
                    title: '消息',
                    msg: '失败',
                    showType: 'show'
                });
            }
        }
    });
}